watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/dye_desalination/dye_desalination_withRO.py

```diff
@@ -237,7 +259,10 @@ def set_operating_conditions(m):
     solve(m.fs.feed, checkpoint="solve feed block")
 
     # pretreatment
-    prtrt.wwtp.load_parameters_from_database(use_default_removal=True)
+    if include_pretreatment == True:
```

Comment:
Instead of having to pass `include_pretreatment` through each function, you could check if pretreatment exists on the model with `hasattr`.